File: android/src/main/java/expo/modules/adaptiveglass/ExpoAdaptiveGlassLensModule.kt
package expo.modules.adaptiveglass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// own module, older Expo SDKs allow one view per module
class ExpoAdaptiveGlassLensModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdaptiveGlassLens")

    View(GlassLensView::class) {
      Events("onTabSelect")

      Prop("selectedIndex") { view: GlassLensView, value: Int -> view.selectedIndex = value }
      Prop("lensStyle") { view: GlassLensView, value: String -> view.lensStyle = value }
      Prop("refraction") { view: GlassLensView, value: Boolean -> view.refraction = value }
      Prop("tintColor") { view: GlassLensView, value: Int? -> view.tint = value }
      Prop("tintScheme") { view: GlassLensView, value: String -> view.tintScheme = value }

      OnViewDidUpdateProps { view: GlassLensView ->
        view.propsDidUpdate()
      }
    }
  }
}

File: ios/ExpoAdaptiveGlassLensModule.swift
import ExpoModulesCore

// separate module: older SDKs allow one view per module
public class ExpoAdaptiveGlassLensModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdaptiveGlassLens")

    View(GlassLensView.self) {
      Events("onTabSelect")

      Prop("selectedIndex") { (view, value: Int) in view.selectedIndex = value }
      Prop("lensStyle") { (view, value: String) in view.lensStyle = value }
      Prop("restPill") { (view: GlassLensView, value: Bool) in view.restPill = value }
      Prop("pillColor") { (view: GlassLensView, value: UIColor?) in view.pillColor = value }
      Prop("tintColor") { (view, value: UIColor?) in view.glassTint = value }
      Prop("tintScheme") { (view, value: String) in view.tintScheme = value }
      Prop("refraction") { (view, value: Bool) in view.refraction = value }

      OnViewDidUpdateProps { view in
        view.propsDidUpdate()
      }
    }
  }
}
